edgelet/edgelet-core/src/settings.rs

```diff
@@ -424,6 +424,23 @@ pub enum ProvisioningType {
     External(External),
 }
 
+impl ProvisioningType {
+    pub fn name(&self) -> &str {
+        match self {
+            ProvisioningType::Manual(manual) => match manual.authentication_method() {
```

Comment:
Can we add the dynamic_provisioning and always_reprovision_on_startup (see https://github.com/Azure/iotedge/pull/3435)?

Reply:
This should give us a pretty good idea on how customers are provision edge devices.

Reply:
Done.